proposal/package-importer.d.ts.md

```diff
@@ -0,0 +1,386 @@
+`string` and a [previous url] `previousUrl`:
+
+- Let `url` be the result of [parsing `string` as a URL][parsing a url] with
+  `base` as the base URL. If this returns a failure, throw that failure.
+- If `url`'s scheme is not `pkg`, return null.
+- Let `resolved` be the result of [resolving a `pkg` URL] with `url` and `previousURL`
+- If `resolved` is null, return null.
+- Let `text` be the contents of the file at `resolved`.
+- Let `syntax` be:
+  - "scss" if `url` ends in `.scss`.
+  - "indented" if `url` ends in `.sass`.
+  - "css" if `url` ends in `.css`.
+  > The algorithm for [resolving a `file:` URL](../spec/modules.md#resolving-a-file-url)
+  > guarantees that `url` will have one of these extensions.
+- Return `text`, `syntax`, and `resolved`.
+
+[parsing a URL]: https://url.spec.whatwg.org/#concept-url-parser
+[resolving a `pkg` URL]: #resolving-a-pkg-url
+
+> Note that this algorithm does not automatically resolve index files, partials
+> or extensions, except where specified. When using the `pkg:` url scheme,
+> authors need to explicitly use the fully resolved filename.
+
+## Procedures
+
+### Resolving a `pkg` URL
+
+This algorithm takes a URL with scheme `pkg`, and an optional URL `previousURL`. It returns a canonical file path or null.
+
+- Let `fullPath` be `url`'s path.
+- Let `packageName` be the result of [resolving a package name], and `subPath`
+  be the path without the `packageName`.
+- Let `packageRoot` be the result of [resolving the root directory for a
+  package].
+- Let `packageManifest` be the result of loading the `package.json` at
+  `packageRoot`.
+- If `packageManifest` is not set, throw an error.
+- Let `resolved` be the result of [Resolving package exports] to resolve `fullPath`
+  with the `sass` condition set, using `packageManifest`.
+  - If `resolved` has the scheme `file:` and an extension of `sass`, `scss` or
+    `css`, return it.
+  - Otherwise, if `resolved` is not null, throw an error.
+- Let `resolved` be the result of [Resolving package exports] to resolve `fullPath`
+  with the `style` condition set, using `packageManifest`.
+  - If `resolved` has the scheme `file:` and an extension of `sass`, `scss` or
+    `css`, return it.
+  - Otherwise, if `resolved` is not null, throw an error.
+- If `subPath` is empty, return result of [resolving package root values].
+- Let `resolved` be `subPath` resolved relative to `packageRoot`.
+- Return the result of [resolving a file url] with `resolved`.
+
+
+[previous url]: ../accepted/prev-url.d.ts.md
+[Resolving package exports]: #resolving-package-exports
+[resolving package root values]: #resolving-package-root-values
+[resolving a package name]: #resolving-a-package-name
+[parsing a url]: https://url.spec.whatwg.org/#concept-url-parser
+[resolving the root directory for a package]:
+    #resolving-the-root-directory-for-a-package
+[resolving a file url]: ../spec/modules.md#resolving-a-file-url
+
+### Resolving package root values
```

Comment:
Style nit: it's a little easier to follow if sub-procedures are defined in the order they're invoked in the parent procedure.